### PR TITLE
add eth_getQuorumPayload

### PR DESCRIPTION
**Pull request by pinges**
This PR add the eth_getQuorumPayload RPC

Signed-off-by: Stefan Pingel <stefan.pingel@consensys.net>

### CODE REVIEW

File: container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/ContainerTests.java [hunk 1]
```diff
@@ -216,7 +232,7 @@ public void contractShouldBeDeployedOnlyToBesuNode()
     ethFilterSubscription.addSingleTopic(eventEncoded);
 
     // Generate a value to insert into the log
-    final String logValue = generateHexString((1234567L));
+    final String logValue = generate64BytesHexString((1234567L));
```
Comment (macfarla) [inline container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/ContainerTests.java:235]: do we need the double (())

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java [hunk 2]
```diff
@@ -85,6 +85,7 @@
   ETH_GET_MINER_DATA_BY_BLOCK_HASH("eth_getMinerDataByBlockHash"),
   ETH_GET_MINER_DATA_BY_BLOCK_NUMBER("eth_getMinerDataByBlockNumber"),
   ETH_GET_PROOF("eth_getProof"),
+  ETH_GET_QUORUM_PAYLOAD("eth_getQuorumPayload"),
```
Comment (macfarla) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:88]: I know it's called eth_getQuorum... but would it make more sense for the constant to be GOQUORUM_... and put it with the GQ ones?

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/goquorum/GoQuorumEthGetQuorumPayloadTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            new JsonRpcRequest("2.0", "eth_getQuorumPayload", new String[] {hexString}));
+
+    final JsonRpcResponse response = method.response(request);
+    assertThat(response).isInstanceOf(JsonRpcErrorResponse.class);
+    assertThat(response.toString()).contains("INVALID_PARAMS");
+  }
+
+  @Test
+  public void requestNonHexString() {
+    final String hexString = Bytes.wrap(new byte[63]).toHexString() + "f";
```
Comment (macfarla) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/goquorum/GoQuorumEthGetQuorumPayloadTest.java:124]: suggest renaming hexString to nonHexString or invalidHexString

File: container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java [hunk 4]
```diff
@@ -138,10 +139,16 @@ public static String sendLogEventAndReturnTransactionHash(
...
-    final StringBuilder randomValue = new StringBuilder(Long.toHexString(number));
+  public static String generate64BytesHexString(final long number) {
+    final String str = Long.toHexString(number);
 
-    while (randomValue.length() < 64) {
+    return prependZeroesToHexString(str, 64);
+  }
+
+  @NotNull
+  public static String prependZeroesToHexString(final String hexString, final int lenRequested) {
```
Comment (macfarla) [inline container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java:149]: prependZeroesToPadHexStringToGivenLength ?

File: container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java [hunk 5]
```diff
@@ -138,10 +139,16 @@ public static String sendLogEventAndReturnTransactionHash(
...
+    final StringBuilder randomValue = new StringBuilder(hexString);
```
Comment (macfarla) [inline container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java:150]: there's no randomness here, is there? hexStringToPrependZeroes ?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java (hunk 2 above)
Comment (macfarla) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:88]: maybe just like you've done with the class name - GoQuorumEthGetQuorumPayload

**Review by macfarla — COMMENTED** [review]
some comments

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java (hunk 2 above)
Comment (mark-terry) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:88]: I think I like this.

File: container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/ContainerTests.java (hunk 1 above)
Comment (pinges) [reply container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/ContainerTests.java:235]: Done

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java (hunk 2 above)
Comment (pinges) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/RpcMethod.java:88]: Done

File: container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java (hunk 4 above)
Comment (pinges) [reply container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java:149]: Done

File: container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java (hunk 5 above)
Comment (pinges) [reply container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java:150]: Done

**Review by macfarla — APPROVED** [review]
LGTM

**Pull request merged**